torch/csrc/jit/codegen/cuda/executor.cpp

```diff
@@ -225,6 +225,7 @@ void FusionExecutor::compileFusion(
         block_size > 0, "launch param inferred block size < 0");
   }
 
+  block_size_high_water_mark = block_size.has_value() ? block_size.value() : 1;
```

Comment:
nitpick: I think this is safe as compileFusion/lowering only gets called once, but since we already have `block_size_high_water_mark` initialized to 1, can we change this to vvv
```
if (block_size.has_value()) {
  block_size_high_water_mark = block_size.has_value();
}
```